### PR TITLE
Edit member pages #136724739

### PR DESCRIPTION
**Pull request by patmbolger**
PT Story: https://www.pivotaltracker.com/story/show/136724739

Changes proposed in this pull request:
1. Add ability to edit member (static) pages
2. Ckeditor::Picture class enhanced to accommodate images for member pages (in addition to company images)
3. All member pages converted to HTML (from HAML)

Screenshots (Optional):
screen shot 2017-04-03 at 4 12 16 pm
---
screen shot 2017-04-03 at 4 12 35 pm
---
screen shot 2017-04-03 at 4 12 50 pm

Ready for review:
@weedySeaDragon 
@thesuss

### CODE REVIEW

File: app/views/shf_documents/contents_edit.html.haml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+.post-title-divider
+  %span
+
+
+.entry-content.wpcf7
+  = render 'contents_form'
+
+%br
+  .center
+    = link_to 'Cancel', contents_show_path(@page),
```
Comment (weedySeaDragon) [inline app/views/shf_documents/contents_edit.html.haml]: Needs Iocalization: `t('cancel')`

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]
Nice work, Pat.  
I just saw that 1 little t() that is needed.  Other than that, it looks good.

**Comment by thesuss** [comment]
Now, how weird is that? 
skarmavbild 2017-04-04 kl 14 31 12 1

File: app/controllers/shf_documents_controller.rb [hunk 2]
```diff
@@ -2,11 +2,12 @@ class ShfDocumentsController < ApplicationController
 
   before_action :set_shf_document, only:  [ :show, :edit, :update, :destroy ]
   before_action :authorize_shf_doc, only: [ :show, :edit, :update, :destroy ]
-  before_action :authorize_shf_doc_class, only: [:index, :new, :create, :historika_meeting_minutes ]
```
Comment (thesuss) [inline app/controllers/shf_documents_controller.rb:5]: `:historika_meeting_minutes` which I assume is "protokoll" is missing, is it not needed anymore?

File: app/views/application/_navigation_member_pages.html.haml [hunk 3]
```diff
@@ -1,12 +1,5 @@
...
-  %ul.sub-menu
-
-    %li.menu-item
-      = link_to t('menus.nav.members.shf_meeting_minutes'), shf_documents_path
-
-    - HighVoltage.page_ids.each do |page|
-      - unless page == 'index'
-        %li.menu-item
-          = link_to page.capitalize, page_path(page)
+  = render partial: 'shf_documents/documents_list',
```
Comment (thesuss) [inline app/views/application/_navigation_member_pages.html.haml:4]: Isnt this one listing pages? Documents would be pdf's..

File: config/locales/en.yml [hunk 4]
```diff
@@ -523,13 +523,20 @@ en:
   shf_documents:
 
     new_shf_document: &new_shf_document New SHF Document
+    new_shf_minutes:  &new_shf_minutes Add Meeting Minutes Document
     edit_shf_document: Edit the SHF document details
     view_shf_document: View the SHF document details
     all_shf_documents: All SHF Documents
+    all_shf_minutes: All Meeting Minutes
+    edit_shf_document_contents: Edit the SHF document contents
```
Comment (thesuss) [inline config/locales/en.yml]: This is confusing... 
Page or Document or "minute" ?

File: config/locales/sv.yml [hunk 5]
```diff
@@ -524,13 +524,20 @@ sv:
   shf_documents:
 
     new_shf_document: &new_shf_document Nytt dokument
+    new_shf_minutes:  &new_shf_minutes Lägg till mötesprotokoll Document
```
Comment (thesuss) [inline config/locales/sv.yml]: remove `Document`

File: config/locales/sv.yml [hunk 6]
```diff
@@ -524,13 +524,20 @@ sv:
...
     edit_shf_document: Redigera info om dokument
     view_shf_document: Information om dokument
     all_shf_documents: Alla uppladdade dokument
+    all_shf_minutes: Alla mötesprotokoll
+    edit_shf_document_contents: Redigera SHF dokumentinnehållet
```
Comment (thesuss) [inline config/locales/sv.yml]: Where is this? You cannot edit a pdf. 
"Redigera sida" ?

File: config/locales/sv.yml [hunk 7]
```diff
@@ -524,13 +524,20 @@ sv:
...
 
     invalid_upload_type: *invalid_upload_type
     file_too_large: *file_too_large_5MB
 
+    contents_access_error: 'Det uppstod ett fel att komma åt filen: %{message}'
+
+    missing_special_letters: 'I dokumenttitlar ovan, bokstäverna å, ä och ö visas som a, a och o, respektive.'
```
Comment (thesuss) [inline config/locales/sv.yml]: lol 

"Vi beklagar, men för tillfället kan systemet inte visa åäö i undersidornas titlar."

File: app/controllers/shf_documents_controller.rb (hunk 2 above)
Comment (patmbolger) [reply app/controllers/shf_documents_controller.rb:5]: There was no action by that name so I removed this reference.

File: app/views/application/_navigation_member_pages.html.haml (hunk 3 above)
Comment (patmbolger) [reply app/views/application/_navigation_member_pages.html.haml:4]: This lists 1) the "meeting minutes" link (which goes to the index view of the meeting minutes documents), and 2) all of the static pages with links created by the HighVoltage gem.

File: config/locales/en.yml (hunk 4 above)
Comment (patmbolger) [reply config/locales/en.yml]: I agree.  I removed "SHF". 

This is used for the button text that appears in the upper-right hand corner when viewing a "static" member page.  (see 2nd screenshot above)

File: config/locales/sv.yml (hunk 5 above)
Comment (patmbolger) [reply config/locales/sv.yml]: Done

File: config/locales/sv.yml (hunk 6 above)
Comment (patmbolger) [reply config/locales/sv.yml]: See related comment above.

File: config/locales/sv.yml (hunk 7 above)
Comment (patmbolger) [reply config/locales/sv.yml]: Fixed.

File: app/views/application/_navigation_member_pages.html.haml (hunk 3 above)
Comment (thesuss) [reply app/views/application/_navigation_member_pages.html.haml:4]: Ah!

File: app/views/shf_documents/_documents_list.html.haml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+%ul{ class: ul_class }
+
+  %li{ class: li_class }
+    = link_to t('menus.nav.members.shf_meeting_minutes'), shf_documents_path
+
+  - HighVoltage.page_ids.each do |page|
+    %li{ class: li_class }
+      = link_to page.capitalize, contents_show_path(page: page)
```
Comment (thesuss) [inline app/views/shf_documents/_documents_list.html.haml:8]: Aha, I see now. It just got cleaned up a bit =)

**Review by weedySeaDragon — APPROVED** [review]

**Comment by weedySeaDragon** [comment]
The build failed due to the intermittent test fails of doing address geocoding.  sometimes the tests return one set of coordinates, sometimes another.  It almost seems like sometimes it's calling the Google maps API, but sometimes it's getting coordinates another way. (E.g. if you loaded google maps on your browser and did a lookup).  Hmmm.

**Pull request merged**